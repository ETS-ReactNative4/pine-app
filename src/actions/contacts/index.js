export * from './add';
export * from './load';
export * from './markAsRead';
export * from './markAsUnread';
export * from './remove';
export * from './removeAll';
export * from './save';
export * from './sync';
export * from './setLastMessage';
export * from './updateProfiles';
