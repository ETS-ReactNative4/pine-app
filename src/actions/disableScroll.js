export const DISABLE_SCROLL = 'DISABLE_SCROLL';

/**
 * Action to disable or enable scrolling and swiping on the home screen.
 * This is used when previewing a transaction using Peek and Pop (3D Touch).
 */
export const disableScroll = (disable = true) => {
  return {
    type: DISABLE_SCROLL,
    disable
  };
};
