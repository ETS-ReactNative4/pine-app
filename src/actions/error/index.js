export * from './dismiss';
export * from './handle';
