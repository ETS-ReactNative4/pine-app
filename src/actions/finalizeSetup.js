import { add as addDeviceTokenToPine } from './pine/deviceTokens/add';
import { load as loadPineCredentials } from './pine/credentials';
import { sync as syncApp } from './sync';
import { ready as onReady } from './ready';

export const FINALIZE_SETUP_REQUEST = 'FINALIZE_SETUP_REQUEST';
export const FINALIZE_SETUP_SUCCESS = 'FINALIZE_SETUP_SUCCESS';
export const FINALIZE_SETUP_FAILURE = 'FINALIZE_SETUP_FAILURE';

const finalizeSetupRequest = () => {
  return {
    type: FINALIZE_SETUP_REQUEST
  };
};

const finalizeSetupSuccess = () => {
  return {
    type: FINALIZE_SETUP_SUCCESS
  };
};

const finalizeSetupFailure = (error) => {
  return {
    type: FINALIZE_SETUP_FAILURE,
    error
  };
};

/**
 * Action to finalize the app setup.
 */
export const finalizeSetup = () => {
  return (dispatch) => {
    dispatch(finalizeSetupRequest());

    return dispatch(loadPineCredentials())
      .then(() => {
        return dispatch(addDeviceTokenToPine()).catch(() => {
          /**
           * Suppress errors as the account has still been created and
           * the device token will be added again every time at startup.
           */
        });
      })
      .then(() => {
        return dispatch(syncApp());
      })
      .then(() => {
        dispatch(finalizeSetupSuccess());
        dispatch(onReady());
      })
      .catch((error) => {
        dispatch(finalizeSetupFailure(error));
        throw error;
      });
  };
};
