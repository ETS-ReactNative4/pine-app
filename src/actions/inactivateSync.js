export const INACTIVATE_SYNC = 'INACTIVATE_SYNC';

export const inactivateSync = () => ({
  type: INACTIVATE_SYNC
});
