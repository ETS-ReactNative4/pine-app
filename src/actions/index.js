export * from './load';
export * from './reset';
export * from './navigateWithReset';
