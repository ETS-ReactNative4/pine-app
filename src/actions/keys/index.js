export * from './add';
export * from './load';
export * from './remove';
