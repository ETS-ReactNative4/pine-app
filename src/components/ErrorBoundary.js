import React, { Component } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { handle as handleError } from '../actions/error';

export default class ErrorBoundary extends Component {
  componentDidCatch(error) {
    const dispatch = this.props.dispatch;
    dispatch(handleError(error));
  }

  render() {
    return (
      <View {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node,
  dispatch: PropTypes.func
};
