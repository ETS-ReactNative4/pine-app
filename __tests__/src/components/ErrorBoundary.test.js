import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import { HANDLE_ERROR } from '../../../src/actions';
import ErrorBoundary from '../../../src/components/ErrorBoundary';

const dispatchMock = jest.fn();

jest.mock('../../../src/containers/ErrorModalContainer', () => 'ErrorModalContainer');

describe('ErrorBoundary', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ErrorBoundary dispatch={dispatchMock}>
        <Text>496f7017-d1ef-430c-a2da-b2a4ac094798</Text>
      </ErrorBoundary>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('#componentDidCatch(error)', () => {
    it('dispatches an action of type HANDLE_ERROR with the error', () => {
      const error = new Error('11a8def9-f495-4cc9-aba4-181284bb2280');

      const errorBoundary = new ErrorBoundary({
        dispatch: dispatchMock
      });

      errorBoundary.componentDidCatch(error);

      expect(dispatchMock).toHaveBeenCalledWith({
        type: HANDLE_ERROR,
        error
      });
    });
  });
});
