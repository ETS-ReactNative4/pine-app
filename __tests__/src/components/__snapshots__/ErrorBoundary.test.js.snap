// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders correctly 1`] = `
<View
  dispatch={[MockFunction]}
>
  <Text>
    496f7017-d1ef-430c-a2da-b2a4ac094798
  </Text>
  <ErrorModalContainer />
</View>
`;
